doc/ROADMAP.rst.txt

```diff
@@ -0,0 +1,377 @@
+reimplement parts in Cython?
+
+    - Small matrices are slower than PySparse, needs fixing
+
+There are a lot of formats.  These should be kept, but
+improvements/optimizations should go into CSR/CSC, which are the preferred
+formats.  LIL may be the exception, it's inherently inefficient.  It could be
+dropped if DOK is extended to support all the operations LIL currently
+provides.
+
+Don't emulate np.matrix behavior, drop 2-D?
+
+
+sparse.csgraph
+``````````````
+This module is in good shape.
+
+
+sparse.linalg
+`````````````
+Arpack is in good shape.
+
+isolve:
+
+    - callback keyword is inconsistent 
+    - tol keyword is broken, should be relative tol
+    - Fortran code not re-entrant (but we don't solve, maybe re-use from
+      PyKrilov)
+
+dsolve:
+
+    - add sparse Cholesky or incomplete Cholesky
+    - look at CHOLMOD
+
+
+Ideas for new features:
+
+    - Wrappers for PROPACK for faster sparse SVD computation.
+
+spatial
+```````
+KDTree/cKDTree and the QHull wrappers are in good shape.  The distance module
+needs bug fixes in the distance metrics, and distance_wrap.c needs to be
+cleaned up (maybe rewrite in Cython).
+
+
+special
+```````
+special has a lot of functions that need improvements in precision.  All
+functions that are also implemented in mpmath can be tested against mpmath, and
+should match well.
+
+Things not in mpmath:
+
+  - cdflib
+  - <Pauli checks> some others
+
+
+stats
+`````
+This is a large module with by far the most open issues.  It has improved a lot
+over the past few releases, but more cleanup and rewriting of functions is
+needed.  The Statistics Review milestone on Github gives a reasonable overview
+of which functions need checking, documentation and tests. 
+
+``stats.distributions`` is in good shape.
+
+All hypothesis tests should get a keyword 'alternative' where applicable (see
+``stats.kstest`` for an example).
+
+``gaussian_kde`` is in good shape but limited. It should not be expanded
+probably, this fits better in statsmodels (which already has a lot more KDE
+functionality).
+
+``stats.mstats`` is a useful module for worked with data with missing values.
+One problem it has though is that in many cases the functions have diverged
+from their counterparts in `scipy.stats`.  The ``mstats`` functions should be
+updated so that the two sets of functions are consistent.
+
+
+weave
+`````
+This module is deprecated and will be removed for Scipy 1.0.  It has been
+packaged as a separate package "weave", which users that still rely on the
+functionality provided by ``scipy.weave`` should use.
+
+Also note that this is the only module that was not ported to Python 3.
+
+
+New modules under discussion
+----------------------------
+
+diff
+````
+Currently Scipy doesn't provide much support for numerical differentiation.
+A new ``scipy.diff`` module for that is discussed in
+https://github.com/scipy/scipy/issues/2035.  There's also a fairly detailed
+GSoC proposal to build on, see LINK.
```

Comment:
Might mention `approx_derivative` which is for now private in scipy.optimize. I think it could serve as a basis for scipy.diff rather than (or as an alternative to) numdifftools.